contracts/ism-multisig/src/contract.rs

```diff
@@ -192,17 +75,35 @@ pub fn query(deps: Deps, _env: Env, msg: ISMQueryMsg) -> Result<Binary, Contract
 
     match msg {
         ModuleType => Ok(to_binary(&ISMType::Owned)?),
+        // TODO: ask what is stand for?
+        Verify {
+            metadata: raw_metadata,
+            message: raw_message,
+        } => {
+            let metadata: MessageIdMultisigIsmMetadata = raw_metadata.into();
+            let message: Message = raw_message.into();
+
+            let threshold = THRESHOLD.load(deps.storage, message.origin_domain.into())?;
+            let validators = VALIDATORS.load(deps.storage, message.origin_domain.into())?;
+
+            let mut signatures: Vec<Binary> = Vec::new();
+            for i in 0..metadata.signatures_len().unwrap() {
+                signatures.push(metadata.signature_at(i))
+            }
 
-        Verify { metadata, message } => {
-            let config = CONFIG.load(deps.storage)?;
-
-            let digest = sha256_digest(Binary::from(message))?;
+            let mut success: u8 = 0;
+            for validator in validators.0 {
+                for signature in &signatures {
+                    let verified = deps
+                        .api
+                        .secp256k1_verify(&message.body, signature, &validator.signer_pubkey)
+                        .unwrap();
 
-            let verified = deps
-                .api
-                .secp256k1_verify(&digest, &metadata, &config.owner_pubkey)?;
+                    success += if verified { 1 } else { 0 }
```

Comment:
```suggestion
                    success += verified as u8
```

according to https://doc.rust-lang.org/std/primitive.bool.html, boolean type of rust is guaranteed to convert 0 / 1